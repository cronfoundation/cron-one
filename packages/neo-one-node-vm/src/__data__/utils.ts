import { utils } from '@neo-one/client-common';
import { StorageItem } from '@neo-one/node-core';

// tslint:disable-next-line no-any
export const verifyBlockchainSnapshot = (blockchain: any) => {
  Object.entries(blockchain).forEach(([name, obj]) => {
    if (typeof obj === 'object' && obj !== null) {
      Object.entries(obj).forEach(([mockName, maybeMock]) => {
        if (maybeMock != undefined && maybeMock.mock != undefined) {
          expect(`${name}.${mockName}`).toMatchSnapshot();
          expect(maybeMock.mock.calls).toMatchSnapshot();
        }
      });
    }
  });
};

// tslint:disable-next-line no-any
export const verifyListeners = (listeners: any) => {
  // tslint:disable-next-line no-any
  Object.values(listeners).forEach((func: any) => {
    expect(func.mock.calls).toMatchSnapshot();
  });
};

export const expectItemBNEquals = (item: StorageItem, value: string) => {
  expect(utils.fromSignedBuffer(item.value).toString(10)).toEqual(value);
};

// tslint:disable-next-line:no-any readonly-array
export const badSeen = (...args: any[]) => new Set(args);
