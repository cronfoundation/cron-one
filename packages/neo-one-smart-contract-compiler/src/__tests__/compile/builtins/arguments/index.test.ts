import { helpers } from '../../../../__data__';
import { DiagnosticCode } from '../../../../DiagnosticCode';

describe('IArguments', () => {
  test('cannot be implemented', async () => {
    helpers.compileString(
      `
      class MyArray implements IArguments {
      }
    `,
      { type: 'error' },
    );
  });

  test.skip('cannot be referenced', async () => {
    helpers.compileString(
      `
      const y = () => {
        const x = arguments;
      }
    `,
      { type: 'error', code: DiagnosticCode.InvalidBuiltinReference },
    );
  });
});
