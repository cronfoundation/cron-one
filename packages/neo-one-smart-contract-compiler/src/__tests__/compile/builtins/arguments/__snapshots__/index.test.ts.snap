// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IArguments cannot be implemented 1`] = `
"snippetCode.ts (2,13): Class 'MyArray' incorrectly implements interface 'IArguments'.
  Property '[one0]' is missing in type 'MyArray' but required in type 'IArguments'.

      1 | 
    > 2 |       class MyArray implements IArguments {
        |             ^
      3 |       }
      4 |     
"
`;

exports[`IArguments cannot be referenced 1`] = `
"snippetCode.ts (3,19): Builtins cannot be referenced

      1 | 
      2 |       const y = () => {
    > 3 |         const x = arguments;
        |                   ^
      4 |       }
      5 |     
"
`;
